src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -122,6 +128,37 @@ private static SignatureType GetSignatureType(SignerInfo signer)
             // TODO: Change this to use the new attributes that justin is adding.
             return SignatureType.Author;
         }
+        private static Timestamp GetTimestamp(SignerInfo signer)
+        {
+            var authorUnsignedAttributes = signer.UnsignedAttributes;
+            var timestampCms = new SignedCms();
+
+            foreach (var attribute in authorUnsignedAttributes)
+            {
+                if (string.Equals(attribute.Oid.Value, Oids.SignatureTimeStampTokenAttributeOid))
+                {
+                    timestampCms.Decode(attribute.Values[0].RawData);
+
+                    if (Rfc3161TimestampVerificationUtility.TryReadTSTInfoFromSignedCms(timestampCms, out var tstInfo))
+                    {
+                        const long TicksPerMicrosecond = 10;
+
+                        var accuracy = tstInfo.AccuracyInMicroseconds;
+                        var accuracyTimeSpan = TimeSpan.Zero;
+                        if (accuracy != null)
+                        {
+                            accuracyTimeSpan = TimeSpan.FromTicks(accuracy.Value * TicksPerMicrosecond);
+                        }
+
+                        return new Timestamp(timestampCms.SignerInfos[0],
```

Comment:
the constructor should only get the `SignerInfos[0]` and the `tstInfo`. Let's keep all of the computations related to the timestamp in their our class.

Reply:
Ok, will refactor to make that change.